joint_limits/include/joint_limits/joint_limits.hpp

```diff
@@ -0,0 +1,163 @@
+ *   upper --> max_position
+ *   velocity --> max_velocity
+ *   effort --> max_effort
+ */
+struct JointLimits
+{
+  JointLimits()
+  : min_position(std::numeric_limits<double>::quiet_NaN()),
+    max_position(std::numeric_limits<double>::quiet_NaN()),
+    max_velocity(std::numeric_limits<double>::quiet_NaN()),
+    max_acceleration(std::numeric_limits<double>::quiet_NaN()),
+    max_jerk(std::numeric_limits<double>::quiet_NaN()),
+    max_effort(std::numeric_limits<double>::quiet_NaN()),
+    has_position_limits(false),
+    has_velocity_limits(false),
+    has_acceleration_limits(false),
+    has_jerk_limits(false),
+    has_effort_limits(false),
+    angle_wraparound(false)
+  {
+  }
+
+  double min_position;
+  double max_position;
+  double max_velocity;
+  double max_acceleration;
+  double max_jerk;
+  double max_effort;
+
+  bool has_position_limits;
+  bool has_velocity_limits;
+  bool has_acceleration_limits;
+  bool has_jerk_limits;
+  bool has_effort_limits;
+  bool angle_wraparound;
+
+  std::string to_string()
+  {
+    std::stringstream ss_output;
+
+    if (has_position_limits)
+    {
+      ss_output << "  position limits: "
+                << "[" << min_position << ", " << max_position << "]\n";
+    }
+    if (has_velocity_limits)
+    {
+      ss_output << "  velocity limit: "
+                << "[" << max_velocity << "]\n";
+    }
+    if (has_acceleration_limits)
+    {
+      ss_output << "  acceleration limit: "
+                << "[" << max_acceleration << "]\n";
+    }
+    if (has_jerk_limits)
+    {
+      ss_output << "  jerk limit: "
+                << "[" << max_jerk << "]\n";
+    }
+    if (has_effort_limits)
+    {
+      ss_output << "  effort limit: "
+                << "[" << max_effort << "]\n";
+    }
+    if (angle_wraparound)
+    {
+      ss_output << "  angle wraparound is active.";
+    }
+
+    return ss_output.str();
+  }
+
+  std::string debug_to_string()
+  {
+    std::stringstream ss_output;
+
+    ss_output << "  has position limits: " << (has_position_limits ? "true" : "false") << "["
+              << min_position << ", " << max_position << "]\n";
+    ss_output << "  has velocity limits: " << (has_velocity_limits ? "true" : "false") << "["
+              << max_velocity << "]\n";
+    ss_output << "  has acceleration limits: " << (has_acceleration_limits ? "true" : "false")
+              << " [" << max_acceleration << "]\n";
+    ss_output << "  has jerk limits: " << (has_jerk_limits ? "true" : "false") << "[" << max_jerk
+              << "]\n";
+    ss_output << "  has effort limits: " << (has_effort_limits ? "true" : "false") << "["
+              << max_effort << "]\n";
+    ss_output << "  angle wraparound: " << (angle_wraparound ? "true" : "false");
+
+    return ss_output.str();
+  }
+};
+
+/**
+ * SoftJointLimits stores values from the `<safety_controller>` tag of URDF.
+ * For details about meaning of each variable check here: https://wiki.ros.org/urdf/XML/joint
```

Comment:
it's ok to have this doc but I think a ROS1 wiki page may be a little confusing to people. Probably better to copy over the description of the members here